diff -urN lincity-1.11/Makefile lincity-1.11.new/Makefile
--- lincity-1.11/Makefile	Fri Feb 26 05:17:38 1999
+++ lincity-1.11.new/Makefile	Fri Oct  6 19:11:31 2000
@@ -1,19 +1,19 @@
 #  Makefile for LinCity   (c)  I J Peters 1995,1996.
 
 #       This is where the binaries for LinCity will be placed.
-LC_BINDIR = /usr/local/bin
+LC_BINDIR = $(DESTDIR)/usr/games
 #          for solaris you might choose...
 #          LC_BINDIR = /opt/lcity/bin
 
 #       This is where the library/data files will be placed.
-LC_LIBDIR = /usr/local/lib/lincity
+LC_LIBDIR = $(DESTDIR)/usr/lib/games/lincity
 #          for solaris you might choose...
 #          LC_LIBDIR = /opt/lcity/lib
 
 # export LC_LIBDIR
 
 #       This is where the manpage will be placed.
-LC_MANDIR = /usr/local/man/man6
+LC_MANDIR = $(DESTDIR)/usr/share/man/man6
 LC_MANSUF = 6
 #          for solaris you might choose...
 #          LC_LIBDIR = /opt/lcity/man/man6
