--- lincity109/Makefile.path	Thu Nov 13 17:24:33 1997
+++ lincity109/Makefile	Thu Nov 13 17:26:43 1997
@@ -1,20 +1,20 @@
 #  Makefile for LinCity   (c)  I J Peters 1995,1996.
 
 #       This is where the binaries for LinCity will be placed.
-LC_BINDIR = /usr/local/bin
+LC_BINDIR = $(INSTALL)/usr/X11R6/bin
 #          for solaris you might choose...
 #          LC_BINDIR = /opt/lcity/bin
 
 #       This is where the library/data files will be placed.
-LC_LIBDIR = /usr/local/lib/lincity
+LC_LIBDIR = $(INSTALL)/usr/lib/lincity
 #          for solaris you might choose...
 #          LC_LIBDIR = /opt/lcity/lib
 
 # export LC_LIBDIR
 
 #       This is where the manpage will be placed.
-LC_MANDIR = /usr/local/man/man6
-LC_MANSUF = 6
+LC_MANDIR = $(INSTALL)/usr/X11R6/man/man1
+LC_MANSUF = 1
 #          for solaris you might choose...
 #          LC_LIBDIR = /opt/lcity/man/man6
 
@@ -23,7 +23,7 @@
 EXTRA_OPTS = -fstrength-reduce -ffast-math
 WARN_OPTS  = -Wall -Wpointer-arith -Wcast-align -Winline
 # DEBUG_OPTS =-g
-CFLAGS = -O2 $(DEBUG_OPTS) $(EXTRA_OPTS) $(WARN_OPTS) -DLIBDIR=\"$(LC_LIBDIR)\"
+CFLAGS = -O2 $(DEBUG_OPTS) $(EXTRA_OPTS) $(WARN_OPTS) -DLIBDIR=\"$(LC_LIBDIR)\" -I/usr/X11R6/include
 # -fno-strength-reduce
 # PROFF = -pg
 # PROFL = -lc
