diff -ur lincity109/help.cxx lincity109.new/help.cxx
--- lincity109/help.cxx	Sun Apr 13 18:44:07 1997
+++ lincity109.new/help.cxx	Sat Jun 17 23:08:15 2000
@@ -4,9 +4,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <strings.h>
-#ifdef SCO
+//#ifdef SCO
 #include <string.h>
-#endif
+//#endif
 #include <ctype.h>
 #ifdef LC_X11
 #include <X11/Xlib.h>
diff -ur lincity109/main.cxx lincity109.new/main.cxx
--- lincity109/main.cxx	Fri Apr 18 18:51:22 1997
+++ lincity109.new/main.cxx	Sat Jun 17 23:05:02 2000
@@ -5,9 +5,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <strings.h>
-#ifdef SCO
+//#ifdef SCO
 #include <string.h>
-#endif
+//#endif
 
 #ifdef LC_X11
 #include <X11/Xlib.h>
@@ -50,7 +50,7 @@
 int prof_countdown=PROFILE_COUNTDOWN;
 #endif
 
-main(int argc,char *argv[])
+int main(int argc,char *argv[])
 {
 	int obx=0,oby=0;
 #ifdef LC_X11
@@ -1724,7 +1724,7 @@
 
 void time_for_year(void)
 {
-	static time_last_year=0;
+	static long time_last_year=0;
 	char s[100];
 	float f;
 	f=(float)(real_time-time_last_year)/1000.0;
diff -ur lincity109/mouse.cxx lincity109.new/mouse.cxx
--- lincity109/mouse.cxx	Mon Apr 14 01:46:27 1997
+++ lincity109.new/mouse.cxx	Sat Jun 17 23:05:59 2000
@@ -4,9 +4,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <strings.h>
-#ifdef SCO
+//#ifdef SCO
 #include <string.h>
-#endif
+//#endif
 
 #ifdef LC_X11
 #include <X11/Xlib.h>
@@ -2061,7 +2061,7 @@
 
 int mt_draw(int x,int y)
 {
-	static flag=0;
+	static int flag=0;
 	static int tlinex[30],tliney[30]; // only need 25
 	int i,j,dx,dy;
 	if (flag==0)
diff -ur lincity109/mps.cxx lincity109.new/mps.cxx
--- lincity109/mps.cxx	Mon Apr 14 02:14:05 1997
+++ lincity109.new/mps.cxx	Sat Jun 17 23:07:17 2000
@@ -15,9 +15,9 @@
 #endif
 #include <math.h>
 #include <strings.h>
-#ifdef SCO
+//#ifdef SCO
 #include <string.h>
-#endif
+//#endif
 #include <unistd.h>
 #include "lctypes.h"
 #include "lin-city.h"
@@ -26,7 +26,7 @@
 
 void mappoint_stats(int x,int y,int button)
 {
-	static xx=0,yy=0,oldbut=0;
+	static int xx=0,yy=0,oldbut=0;
 	char s[100];
 
 	if (x<0 || y<0)
diff -ur lincity109/screen.cxx lincity109.new/screen.cxx
--- lincity109/screen.cxx	Mon Apr 14 01:40:53 1997
+++ lincity109.new/screen.cxx	Sat Jun 17 23:06:31 2000
@@ -17,9 +17,9 @@
 
 #include <math.h>
 #include <strings.h>
-#ifdef SCO
+//#ifdef SCO
 #include <string.h>
-#endif
+//#endif
 #include <unistd.h>
 #include "lctypes.h"
 #include "lin-city.h"
diff -ur lincity109/typeinit.cxx lincity109.new/typeinit.cxx
--- lincity109/typeinit.cxx	Mon Apr 14 01:42:54 1997
+++ lincity109.new/typeinit.cxx	Sat Jun 17 23:07:43 2000
@@ -15,9 +15,9 @@
 #include <vgamouse.h>
 #endif
 #include <strings.h>
-#ifdef SCO
+//#ifdef SCO
 #include <string.h>
-#endif
+//#endif
 #include "lctypes.h"
 #include "lin-city.h"
 #include "typeinit.h"
